File: LID_Framework/Download.cs
using System;
using System.Net;

namespace LID_Framework {
    class Download {
        string outfile;

        public Download() {
            DownloadFromWeb();
        }

        public void DownloadFromWeb() {
            outfile = (@"Files\Bulletins\" + DateTime.UtcNow.ToString("MM-dd-yyyy") + "_Bulletin_Pull.txt").Replace(" ", "");
            DownloadFile("https://www.navcen.uscg.gov/?pageName=iipB12Out", outfile);

            string text = System.IO.File.ReadAllText(outfile);
            text = text.Replace("</p>", "");
            text = text.Replace("<p>", "");
            text = text.Replace("	", "");
            text = text.Replace(".", ". ");
            text = text.Replace(",", ", ");
            text = text.Substring(text.IndexOf("NORTH AMERICAN ICE SERVICE (NAIS)"), text.IndexOf("CANCEL THIS MSG") - text.IndexOf("NORTH AMERICAN ICE SERVICE (NAIS)") + 31);
            System.IO.File.WriteAllText(outfile, text);
        }

        public static void DownloadFile(string Bulletin, string SavedBulletin) {
            WebClient client = new WebClient();
            client.DownloadFile(Bulletin, SavedBulletin);
            client.Dispose();
        }

        //Accessors
        public string GetOutFile() {
            return (outfile);
        }
    }
}

File: LID_Framework/download.py
#!/usr/bin/python3 

from urllib.request import urlopen
import datetime
import sys

#Do a pass arguments thing
	#URL
	#Date/File format
	#Path

#Default URL (Make this an Argument)
url = 'https://www.navcen.uscg.gov/?pageName=iipB12Out'

#Don't need (Is for testing)
file = 'garbagehtml.txt'

#Connection and download, string conversion
data = urlopen(url).read()
data = str(data)

#Writes to the file (For testing)
f = open(file, 'w')
f.write(data)

#Actual Parser itself (Could clean up, just tested each step one at a time)
#Need to add back begginig and end or find different split points
parse = open(file, 'rt').read()
first, parse = parse.split('1.')
parse, second = parse.split('CANCEL THIS')
parse = parse.replace('<p>', '')
parse = parse.replace('</p>', '')
parse = parse.replace('\\n' , '')
parse = parse.replace('\\t' , '')

#For testing
print(parse)

# Date time to name the output file. 
#Pass arguments would be prefered but could have this as backup or defualt
date = datetime.datetime.now()
date = str(date.year) + '-' + str(date.month) + '-' + str(date.day) + '_'
fileout = date + 'bulletin_pull.txt'

#For testing
print(fileout)

#Writes to the Bulletin file
f = open(fileout, 'w')
f.write(parse)

